contracts/Staker.sol

```diff
@@ -526,17 +531,45 @@ contract Staker is Ownable, ReentrancyGuard {
       "You cannot withdraw that much of the specified token; you are not owed it.");
     updatePool(_token);
     uint256 pendingTokens = ((user.amount * pool.tokensPerShare) / 1e12) - user.tokenPaid;
-    token.safeTransferFrom(address(this), msg.sender, pendingTokens);
+    user.tokenRewards += pendingTokens;
     totalTokenDisbursed = totalTokenDisbursed + pendingTokens;
     uint256 pendingPoints = ((user.amount * pool.pointsPerShare) / 1e30) - user.pointPaid;
-    userPoints[msg.sender] = userPoints[msg.sender] + pendingPoints;
+    user.pointRewards += pendingPoints;
     user.amount = user.amount - _amount;
     user.tokenPaid = (user.amount * pool.tokensPerShare) / 1e12;
     user.pointPaid = (user.amount * pool.pointsPerShare) / 1e30;
     pool.token.safeTransfer(address(msg.sender), _amount);
     emit Withdraw(msg.sender, _token, _amount);
   }
 
+  /**
+    Claim accumulated token and point rewards from the Staker.
+    @param _token The asset to claim rewards from.
+   */
+  function claim(IERC20 _token) external nonReentrant {
+    UserInfo storage user = userInfo[_token][msg.sender];
+    PoolInfo storage pool = poolInfo[_token];
+    uint256 pendingTokens;
+    uint256 pendingPoints;
+
+    updatePool(_token);
+    if (user.amount > 0) {
+      pendingTokens = ((user.amount * pool.tokensPerShare) / 1e12) - user.tokenPaid;
+      pendingPoints = ((user.amount * pool.pointsPerShare) / 1e30) - user.pointPaid;
+      totalTokenDisbursed = totalTokenDisbursed + pendingTokens;
+    }
+    uint256 _tokenRewards = user.tokenRewards + pendingTokens;
+    uint256 _pointRewards = user.pointRewards + pendingPoints;
+    token.safeTransferFrom(address(this), msg.sender, _tokenRewards);
```

Comment:
While we are probably safe by using the nonReentrant modifier, I'd like to make sure we still follow the rule "checks -> effects -> interactions" pattern.  My request would be that we move the the safeTransferFrom call below user.tokenRewards, and user.pointRewards being reset to 0 [edit: might as well move under user.tokenPaid, and pointPaid calculations on ln 569].  The tx will still revert if the safeTransfer fails, and this way we can reduce potential vulnerability (however unlikely)